lib/reporters/base.js

```diff
@@ -156,14 +156,8 @@ exports.cursor = {
 };
 
 function showDiff (err) {
-  return err && err.showDiff !== false && sameType(err.actual, err.expected) && err.expected !== undefined;
-}
-
-function stringifyDiffObjs (err) {
-  if (!utils.isString(err.actual) || !utils.isString(err.expected)) {
-    err.actual = utils.stringify(err.actual);
-    err.expected = utils.stringify(err.expected);
-  }
+  return (utils.type(err.diff) === 'function') ||
```

Comment:
so it was agreed that `err.diff` is a *function*?  what's its signature?  can you point me to that discussion?  again, I'm having trouble following / understanding what exactly people want here.

Reply:
Function is used because we need one of two formats for different types of diff as one of possible solutions for this. Perhaps, it will be possible to integrate them in one format, in that case function is not required and this field will be able to be a simple object in this format. In this case this it will be necessary to make utility functions for conversion of results of `diff` library to this universal format. Since in https://github.com/chaijs/chai/issues/1120 already expressed idea of object diff library, it can be done in there.

At the moment I only used existing formats which are provided by `diff` library that simplifies implementation of the user-defined diffs. Signature:
```js
/**
@param {boolean} inline Generate inline diffs if true, generate unified diff otherwise
@return {Object} The same as return value of `diff.structuredPatch` from the `diff` library if `inline` is falsely
                 The same as return value of `diff.diffWordsWithSpace` from the `diff` library if `inline` is truely
*/
function diff(inline) { ... }
```

However, if we are going to make a utility library for generation of diff between objects in the format suitable us

Reply:
Though use of universal format will give nothing -- for generation it is necessary to use different functions. As the choice of a format is carried out by mocha, and generation of diff by assertion library, it is necessary either to use function, or to generate directly 2 kinds of diff, which may be time expensive